src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -405,27 +384,28 @@ protected Date expiration() {
    *
    * @throws IOException If something went wrong.
    */
-  protected void storeLocalRepositoriesInfo() throws IOException {
-    synchronized (LOCAL_REPOSITORIES_INFO_LOCK) {
+  protected static void storeLocalRepositoriesInfo() throws IOException {
+    synchronized (LOCAL_REPOSITORIES_INFO) {
       Files.write(
-          base.resolve(DEFAULT_LOCAL_REPOSITORIES_INFO_FILE),
-          MAPPER.writeValueAsBytes(localRepositoriesInfo));
+          DEFAULT_DIRECTORY.resolve(DEFAULT_LOCAL_REPOSITORIES_INFO_FILE),
```

Comment:
If you change `DEFAULT_DIRECTORY` to be an instance of `Path`, you can also change the type of `DEFAULT_LOCAL_REPOSITORIES_INFO_FILE` as well.